src/main/java/com/atlassian/httpclient/apache/httpcomponents/DefaultHttpClientFactory.java

```diff
@@ -20,7 +20,8 @@
     private final EventPublisher eventPublisher;
     private final ApplicationProperties applicationProperties;
     private final ThreadLocalContextManager threadLocalContextManager;
-    private final Set<ApacheAsyncHttpClient> httpClients = new CopyOnWriteArraySet<ApacheAsyncHttpClient>();
+    // shared http client
+    private static final Set<ApacheAsyncHttpClient> httpClients = new CopyOnWriteArraySet<>();
```

Comment:
we don't need a set now if we create only one instance

Reply:
correct!

Reply:
fixed